contracts/src/staking/StakingManager.sol

```diff
@@ -355,7 +356,18 @@ abstract contract StakingManager is ReentrancyGuard, IStakingManager {
      */
     // solhint-disable-next-line no-empty-blocks
     function resendEndValidatorMessage(bytes32 validationID) external {
-        // TODO: Allow for these messages to be resent.
+        StakingManagerStorage storage $ = _getStakingManagerStorage();
+        Validator memory validator = $._validationPeriods[validationID];
+
+        require(
+            validator.status == ValidatorStatus.PendingRemoved,
+            "StakingManager: Validator not pending removal"
+        );
+
+        bytes memory setValidatorWeightPayload = StakingMessages.packSetSubnetValidatorWeightMessage(
+            validationID, validator.messageNonce, 0
```

Comment:
We are reusing the previous message nonce. Is there any case where we may need to manually set the message nonce? I suppose no additional new messages should be sent after initializeEndValidation.

Reply:
The dynamics of this nonce are specified [here](https://github.com/avalanche-foundation/ACPs/tree/main/ACPs/77-reinventing-subnets#setsubnetvalidatorweighttx). Specifically, the `minNonce` associated with the `validationID` tracked by the P-Chain is incremented whenever a `SetSubnetValidatorWeightTx` is accepted. Since this method is only intended to be used when the `SetSubnetValidatorWeightTx` cannot be accepted due to validator churn, it should be safe to reuse the nonce.